pkg/services/automation/generator.go

```diff
@@ -410,6 +411,16 @@ func GenerateResourceName(url gitproviders.RepoURL) string {
 	return models.ConstrainResourceName(url.RepositoryName())
 }
 
+// GetRandomName generates a random unique name.
```

Comment:
This isn't quite what it does, it generates a random unique string, and appends it to the provided string.

This should probably also limit it to a specific length.

Reply:
Also, you aren't seeding rand. This is a bit nitpicky, and probably fine for these scenarios, but if you ARE using `math/rand` you need to seed it with time.Nano or something like that.

Reply:
Thanks both, addressed this in the latest commit :)